ut/performance_tests.cpp

```diff
@@ -0,0 +1,233 @@
+    const auto template_size = sizeof(result_template) - 1;
+
+    const auto start_pos = index % (template_size - RESULT_SIZE);
+    return std::string_view(&result_template[start_pos], RESULT_SIZE);
+}
+
+std::string_view generate(const ColumnString&, size_t index)
+{
+    // ColumString stores item lengts,and on 1M etnries that builds up to extra 1M bytes,
+    // comparing to 8M bytes of serialized data for ColumnFixedString and ColumUInt64.
+    // So in order to make comparison mode fair, reducing size of data item.
+    return generate_string_view<7>(index);
+}
+
+std::string_view generate(const ColumnFixedString&, size_t index)
+{
+    return generate_string_view<8>(index);
+}
+
+std::string_view generate(const ColumnLowCardinalityT<ColumnString>&, size_t index)
+{
+    return generate_string_view<7>(index);
+}
+
+std::string_view generate(const ColumnLowCardinalityT<ColumnFixedString>&, size_t index)
+{
+    return generate_string_view<8>(index);
+}
+
+template <typename ColumnType>
+auto ValidateColumnItems(const ColumnType & col, size_t expected_items)
+{
+    ASSERT_EQ(expected_items, col.Size());
+    // validate that appended items match expected
+    for (size_t i = 0; i < expected_items; ++i)
+    {
+        SCOPED_TRACE(i);
+
+        ASSERT_EQ(col.At(i), generate(col, i));
+        ASSERT_EQ(col[i], generate(col, i));
+    }
+};
+
+template <typename ColumnType>
+ColumnType InstantiateColumn() {
+    if constexpr (std::is_same_v<ColumnType, ColumnFixedString>) {
+        return ColumnType(8);
+    } else if constexpr (std::is_same_v<ColumnType, ColumnLowCardinalityT<ColumnFixedString>>) {
+        return ColumnType(8);
+    } else {
+        return ColumnType();
+    }
+}
+
+template <typename ColumnType>
+class ColumnPerformanceTest : public ::testing::Test {
+};
+
+TYPED_TEST_CASE_P(ColumnPerformanceTest);
+
+TYPED_TEST_P(ColumnPerformanceTest, SaveAndLoad) {
+    auto column = InstantiateColumn<TypeParam>();
+
+    const size_t ITEMS_COUNT = 1'000'000;
+    const int LOAD_AND_SAVE_REPEAT_TIMES = 10; // run Load() and Save() multiple times to cancel out measurement errors.
+
+    std::cerr << "\n===========================================================" << std::endl;
+    std::cerr << "\t" << ITEMS_COUNT << " items of " << column.Type()->GetName()  << std::endl;
+
+    PausableTimer<std::chrono::microseconds> timer;
+
+    timer.Start();
+    for (size_t i = 0; i < ITEMS_COUNT; ++i)
+    {
+        const auto value = generate(column, i);
+        column.Append(value);
+    }
+
+    EXPECT_EQ(ITEMS_COUNT, column.Size());
+    std::cerr << "Appending:\t" << timer.GetTotalElapsed().count() << " us"
+              << std::endl;
+
+    timer.Reset();
+    timer.Start();
+    EXPECT_NO_FATAL_FAILURE(ValidateColumnItems(column, ITEMS_COUNT));
+    std::cerr << "Accessing (twice):\t" << timer.GetTotalElapsed().count() << " us"
```

Comment:
This is also a heavy operation, that will add up to the measurements.

Reply:
Np, fixed